resqpy/well/_wellbore_marker.py

```diff
@@ -0,0 +1,167 @@
+        wbm_node.set('uuid', str(self.uuid))
+
+        # Citation block
+        if self.title is not None:
+            title = self.title
+        citation = self.model.create_citation(root = wbm_node, title = title, originator = self.originator)
+
+        # Add sub-elements to root node
+        boundary_feature_dict = {
+            'GeologicBoundaryKind': ['fault', 'geobody', 'horizon'],
+            'FluidMarker': ['gas down to', 'gas up to', 'oil down to', 'oil up to', 'water down to', 'water up to'],
+            'FluidContact': ['free water contact', 'gas oil contact', 'gas water contact', 'seal', 'water oil contact']
+        }
+        for k, v in boundary_feature_dict.items():
+            if self.marker_type in v:
+                boundary_kind = k
+                break
+
+        wbm_gb_node = rqet.SubElement(wbm_node, ns['resqml2'] + boundary_kind)
+        wbm_gb_node.set(ns['xsi'] + 'type', ns['xsd'] + 'string')
+        wbm_gb_node.text = str(self.marker_type)
+
+        if self.interpretation_uuid is not None:
+            interp_content_type = 'obj_' + self.marker_type.capitalize() + 'Interpretation'
+            interp_root = self.model.root_for_uuid(uuid = self.interpretation_uuid)
+            self.model.create_ref_node(flavour = 'Interpretation',
+                                       title = rqet.find_tag(rqet.find_tag(interp_root, 'Citation'), 'Title').text,
+                                       uuid = self.interpretation_uuid,
+                                       content_type = interp_content_type,
+                                       root = wbm_node)
+        # Extra metadata
+        if hasattr(self, 'extra_metadata') and self.extra_metadata:
+            rqet.create_metadata_xml(node = wbm_node, extra_metadata = self.extra_metadata)
+
+        if parent_node is not None:
+            parent_node.append(wbm_node)
+
+        return wbm_node
+
+    def _load_from_xml(self, marker_node):
+        """Load attributes from xml.
+
+        This is invoked as part of the init method when an existing uuid is given.
+
+        Returns:
+           [bool]: True if successful
+        """
+
+        assert marker_node is not None
+        # Load XML data
+        uuid_str = marker_node.attrib.get('uuid')
+        if uuid_str:
+            self.uuid = bu.uuid_from_string(uuid_str)
+        citation_tag = rqet.find_nested_tags(root = marker_node, tag_list = ['Citation'])
+        assert citation_tag is not None
+        self.title = rqet.find_tag_text(root = citation_tag, tag_name = 'Title')
+        self.originator = rqet.find_tag_text(root = citation_tag, tag_name = 'Originator')
+
+        for boundary_feature_type in ['GeologicBoundaryKind', 'FluidMarker', 'FluidContact']:
+            found_tag_text = rqet.find_tag_text(root = marker_node, tag_name = boundary_feature_type)
+            if found_tag_text is not None:
+                self.marker_type = found_tag_text
+                break
+
```

Comment:
I think we should assert that self.marker_type is not None here (having set it to None before the for loop)